src/plugins/mocap/include/plugins/mocap/mocap.h

```diff
@@ -0,0 +1,406 @@
+     */
+    struct VisionPositionEstimate {
+        uint64_t time_usec; /**< @brief  Position frame timestamp UNIX Epoch time. */
+        PositionBody position_body; /**< @brief Global Body Position. (m). */
+        AngleBody angle_body; /**< @brief Body angle (rad). */
+        Covariance pose_covariance; /**< @brief Row-major representation of pose 6x6
+                                       ross-covariance matrix upper right triangle. */
+        uint8_t reset_counter; /**< @brief Estimate reset counter. */
+    };
+
+    /**
+     * @brief Motion capture attitude and position type.
+     */
+    struct AttitudePositionMocap {
+        uint64_t time_usec; /**< @brief  Position frame timestamp UNIX Epoch time. */
+        Quaternion q; /**< @brief Attitude quaternion (w, x, y, z order,
+                         zero-rotation is 1, 0, 0, 0) */
+        PositionBody position_body; /**< @brief Body position (NED). */
+        Covariance pose_covariance; /**< @brief Row-major representation of pose 6x6
+                                       cross-covariance matrix upper right triangle. */
+    };
+
+    /**
+     * @brief Odometry information with an external interface type.
+     */
+    struct Odometry {
+        /**
+         * @brief Mavlink frame id
+         */
+        enum class MavFrame {
+            MOCAP_NED = 14, /**< @brief Odometry local coordinate frame of data given
+                               by a motion capture system, Z-down (x: north, y: east, z: down). */
+            LOCAL_FRD = 20, /**< @brief  Forward, Right, Down coordinate frame. This is a local
+                               frame with Z-down and arbitrary F/R alignment (i.e. not
+                               aligned with NED/earth frame). Replacement for
+                               MAV_FRAME_MOCAP_NED, MAV_FRAME_VISION_NED, MAV_FRAME_ESTIM_NED. */
+        };
+
+        uint64_t time_usec; /**< @brief Timestamp (0 to use Backend timestamp). */
+        MavFrame frame_id; /**< @brief Coordinate frame of reference for the pose data. */
+        PositionBody position_body; /**< @brief Body position. */
+        Quaternion q; /**< @brief Quaternion components, w, x, y, z
+                         (1 0 0 0 is the null-rotation). */
+        SpeedBody speed_body; /**< @brief Body linear speed (m/s). */
+        AngularVelocityBody angular_velocity_body; /**< @brief Body angular speed (rad/s). */
+        Covariance pose_covariance; /**< @brief Row-major representation of a 6x6
+                                       pose cross-covariance matrix upper right triangle.
+                                       Leave empty if unknown. */
+        Covariance velocity_covariance; /**< @brief Row-major representation of a 6x6
+                                           velocity cross-covariance matrix upper right
+                                           triangle. Leave empty if unknown. */
+
+        Odometry() :
```

Comment:
I would initialize the fields directly above using `{}` instead of the constructor.